pkg/apis/hive/v1alpha1/clusterdeployment_types.go

```diff
@@ -86,6 +86,9 @@ type ClusterDeploymentStatus struct {
 	// AdminKubeconfigSecret references the secret containing the admin kubeconfig for this cluster.
 	AdminKubeconfigSecret corev1.LocalObjectReference `json:"adminKubeconfigSecret"`
 
+	// ClusterVersion locally stores the remote cluster's clusterversion.status.current.version field
+	ClusterVersion string `json:"clusterVersion"`
+
```

Comment:
Realistically I think treating this as just a string will probably get us into a spot where we need to break API compatibility in the near future. We should make this a struct, the Update struct also has a Payload which looks really interesting for our purposes as well and we should.

Should we pull the FULL ClusterVersionStatus? Or just make our own and copy over parts of it? Looking through https://github.com/openshift/api/blob/master/config/v1/types_cluster_version.go#L75 we're not really interested in the Generation or VersionHash but the Conditions and the AvailableUpdates could be extremely useful. I'd be tempted to re-use their type and copy the whole thing.

Thoughts?

CC @csrwng in case he has ideas as well.

Reply:
I agree, I think there's usefulness in embedding the entire status struct